File: weblayer/browser/java/org/chromium/weblayer_private/ApplicationInfoHelper.java
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.weblayer_private;

import android.content.Context;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;
import android.content.pm.PackageManager.NameNotFoundException;

import org.chromium.base.ContextUtils;
import org.chromium.base.annotations.CalledByNative;
import org.chromium.base.annotations.JNINamespace;

/** Exposes values of metadata tags to native code. */
@JNINamespace("weblayer")
public class ApplicationInfoHelper {
    private ApplicationInfoHelper() {}

    /**
     * Returns the boolean value for a metadata tag in the application's manifest.
     */
    @CalledByNative
    public static boolean getMetadataAsBoolean(String metadataTag, boolean defaultValue) {
        Context context = ContextUtils.getApplicationContext();
        try {
            ApplicationInfo info = context.getPackageManager().getApplicationInfo(
                    context.getPackageName(), PackageManager.GET_META_DATA);
            return info.metaData.getBoolean(metadataTag, defaultValue);
        } catch (NameNotFoundException exception) {
            return defaultValue;
        }
    }
}

File: weblayer/browser/android/application_info_helper.cc
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "weblayer/browser/android/application_info_helper.h"

#include "base/android/jni_android.h"
#include "base/android/jni_string.h"
#include "weblayer/browser/java/base_module_jni/ApplicationInfoHelper_jni.h"

namespace weblayer {

// static
bool GetApplicationMetadataAsBoolean(const std::string& key,
                                     bool default_value) {
  auto* env = base::android::AttachCurrentThread();
  return Java_ApplicationInfoHelper_getMetadataAsBoolean(
      env, base::android::ConvertUTF8ToJavaString(env, key), default_value);
}

}  // namespace weblayer
